client/components/data/query-canonical-theme/index.jsx

```diff
@@ -0,0 +1,37 @@
+/**
+ * External dependencies
+ */
+import React, { PropTypes } from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import QueryTheme from 'components/data/query-theme';
+import { isWpcomTheme, isWporgTheme } from 'state/themes/selectors';
+
+const QueryCanonicalTheme = ( { siteId, themeId, isWpcom, isWporg } ) => (
+	<div>
+		<QueryTheme themeId={ themeId } siteId="wpcom" />
```

Comment:
If I understand correctly if `isWpcom` is `true` we do not need to render `QueryTheme` for wpcom, so maybe a guard would be useful. Same for org.